tripy/tripy/frontend/ops/tensor_initializers.py

```diff
@@ -330,21 +333,32 @@ def arange(
 
         assert tp.allclose(output, tp.Tensor(np.arange(2.3, 0.8, -0.2, dtype=np.float32)))
     """
-    from tripy.common.datatype import int64
+    from tripy.common.datatype import int32, int64
+    from tripy.frontend.tensor import Tensor
 
     if dtype == int64:
         raise_error("Known issue with i64. Arange currently does not work with int64 inputs.")
-    if step == 0:
+
+    if isinstance(step, numbers.Number) and step == 0:
         raise_error("Step in arange cannot be 0.", [])
 
-    size = math.ceil((stop - start) / step)
-    if size <= 0:
+    # math.ceil(a / b) is same as -(-a // b). Don't use math.ceil as start, stop or step can be Tensor.
+    size = 0 - ((start - stop) // step)
+    if isinstance(size, numbers.Number) and size <= 0:
         raise_error(
             "Arange tensor is empty.",
             details=[
                 f"start={start}, stop={stop}, step={step}",
             ],
         )
+
+    if isinstance(size, Tensor):
+        from tripy.frontend.trace.ops.cast import cast
+
+        size = cast(size, int32)
```

Comment:
Can't we express this via dtype constraints?

Reply:
size is computed from start, stop and step which are inputs to this function.  (`size = 0 - ((start - stop) // step)`)

Reply:
But if we require start/stop/step to be `int32` then we won't need to cast here right?

Reply:
These can be of float type, ex: `tp.arange(2.3, 0.8, -0.2)`